### PR TITLE
refactor: feature toggle is now a saga

### PR DESCRIPTION
**Pull request by andreabadesso**
### Acceptance Criteria
- Feature toggle is now a saga

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/utils/wallet.js [hunk 1]
```diff
@@ -458,7 +455,7 @@ const wallet = {
    * @inner
    */
   async resetWalletData() {
```
Comment (pedroferreira1) [inline src/utils/wallet.js:457]: Do you still need this method? I saw that you are swapping all the places to use the redux action instead.

File: src/sagas/helpers.js [hunk 2]
```diff
@@ -1,5 +1,31 @@
...
+  const featureTogglesInitialized = yield select((state) => state.featureTogglesInitialized);
+
+  if (!featureTogglesInitialized) {
+    // Wait until featureToggle saga completed initialization, which includes
+    // downloading the current toggle status for this client.
+    yield take(types.FEATURE_TOGGLE_INITIALIZED);
+  }
+}
+
+export function* checkForFeatureFlag(flag) {
```
Comment (pedroferreira1) [inline src/sagas/helpers.js:22]: Missing docstring

File: src/sagas/wallet.js [hunk 3]
```diff
@@ -380,41 +380,6 @@ export function* fetchTokensMetadata(tokens) {
...
-      featureFlags.removeListener(FeatureFlagEvents.ATOMIC_SWAP_ENABLED, listener);
-    };
-  });
-
-  try {
-    while (true) {
-      const newUseWalletService = yield take(channel);
-      const oldUseWalletService = yield select((state) => state.useWalletService);
-
-      if (oldUseWalletService && oldUseWalletService !== newUseWalletService) {
```
Comment (pedroferreira1) [inline src/sagas/wallet.js:406]: Are you doing this check somewhere? I didn't see it. In the mobile PR it's in this same file

File: src/sagas/wallet.js [hunk 4]
```diff
@@ -183,22 +203,11 @@ export function* startWallet(action) {
...
-  const featureFlagsThread = yield fork(listenForFeatureFlags, featureFlags);
-
-  // Keep track of the forked threads so we can cancel them later. We are currently
-  // using this to start the startWallet saga again during a reload
-  const threads = [
-    walletListenerThread,
-    walletReadyThread,
-    featureFlagsThread
-  ];
+  yield fork(listenForWalletReady, wallet);
```
[the remark this answers is not in the document]
Comment (andreabadesso) [reply src/sagas/wallet.js:214]: FIxed in 64015b9cdeb1826f36deb10d12f31e8db8539479

File: src/sagas/helpers.js (hunk 2 above)
Comment (andreabadesso) [reply src/sagas/helpers.js:31]: Added in 391762d466c93306eb5a706ac7ebf923994bc80e

File: src/utils/wallet.js (hunk 1 above)
Comment (andreabadesso) [reply src/utils/wallet.js]: You're right, we can remove it

Fixed in be94f7c3f44c614130b263300673f8c63a4d200a

File: src/sagas/helpers.js (hunk 2 above)
Comment (pedroferreira1) [reply src/sagas/helpers.js:31]: Missing the return type

**Review by alexruzenhack — APPROVED** [review]

**Pull request merged**